### ISSUE TITLE
Cannot configure credential managers - GitHub desktop for Windows

### ISSUE THREAD

**Issue by zdenko-s** [desktop/desktop#13787]
I would like to use AWS CodeCommit and I should specify aws credential manger in `[credential]` section. I removed [credential] section from `%USERPROFILE%\AppData\Local\GitHubDesktop\app-2.9.6\resources\app\git\etc\gitconfig`. When using "GitHub desktop built git" from command line, everything works fine. (git.exe authenticates using aws credential manager specified in %USERPROFILE%\.gitconfig)
While GitHub desktop UI always asks credentials.
Since credential managers are chained, one specified in user ~/.gitconfig is never called.
Fix:
-either use gitconfig from built in git (%USERPROFILE%\AppData\Local\GitHubDesktop\app-2.9.6\resources\app\git)
-give option to configure GitHub desktop's git configuration

**Comment by sergiou87** [comment]
Hi @zdenko-s! 👋 

I'm sorry you're having this authentication issue. When GitHub Desktop UI asks for the credentials and you enter them, does it work? I mean, are you authenticated and able to pull/push content from the repository?

**Comment by zdenko-s** [comment]
Hi,
Our company policy enforces MFA and I cannot log on using user name/password. Using aws git credential manager, at first, I get temporary access key/secret using MFA authentication. Temporary access key/secret are stored in AWS profile and git credential manager reads it.
Since GitHub desktop asks user name/password and tries to authenticate, it would never work because it is disallowed by AWS security policy.

**Comment by sergiou87** [comment]
Oh! Ok, then it _should_ work if, when asked for credentials, you enter that temporary secret as password. At least, that's how it works with [other services](https://github.com/desktop/desktop/tree/development/docs/integrations). Please, let me know if that helps.

**Comment by zdenko-s** [comment]
No, it will not work. AWS MFA Credential helper uses 3 (three) parameters
`aws_access_key_id, aws_secret_access_key, aws_session_token`
GitHub desktop asks for user name and password. No way to specify 3 parameters.  
It is obvious nobody tested GitHub desktop with AWS CodeCommit with MFA.

**Comment by mat100payette** [comment]
After having contacted Github support, I was led to this issue. I'm honestly a bit shocked that AWS (one of the 3 major cloud providers) integration isn't fully supported by Github Desktop.

I will post here the info I gave to support, as my case is slightly different but might still have something in common with this one.

> Is it known whether or not AWS CodeCommit's auth is supported by Github Desktop (Windows 11) when using git-remote-codecommit?
> 
> 
> I have my repo cloned, and normal cmd git commands work but I cannot fetch origin (or do anything really) from the GUI. For example, when I fetch, a popup appears saying this:
> 
> `git: 'remote-codecommit' is not a git command. See 'git --help'.`
> 
> I would get the same error on cmd when I uninstall git-remote-codecommit from my python env. Is there anything in particular that Github Desktop needs for it to recognize/use that somehow? In case it could help, here is my git config --list output:
> 
> diff.astextplain.textconv=astextplain
> filter.lfs.clean=git-lfs clean -- %f
> filter.lfs.smudge=git-lfs smudge -- %f
> filter.lfs.process=git-lfs filter-process
> filter.lfs.required=true
> http.sslbackend=openssl
> http.sslcainfo=C:/Program Files/Git/mingw64/etc/ssl/certs/ca-bundle.crt
> core.autocrlf=true
> core.fscache=true
> core.symlinks=true
> pull.rebase=false
> credential.helper=manager
> credential.https://dev.azure.com.usehttppath=true
> init.defaultbranch=main
> filter.lfs.clean=git-lfs clean -- %f
> filter.lfs.smudge=git-lfs smudge -- %f
> filter.lfs.process=git-lfs filter-process
> filter.lfs.required=true
> user.name=Oppiie
> user.email=matpayette@hotmail.com
> core.editor="C:\Users\myname\AppData\Local\Programs\Microsoft VS Code\bin\code" --wait
> credential.usehttppath=true
> credential.helper=!aws --profile my-profile codecommit credential-helper $@
> credential.https://git-codecommit.ca-central-1.amazonaws.com/v1/repos/repos.provider=generic
> 
> And here is the config in my cloned repo:
> 
> [core]
> 	repositoryformatversion = 0
> 	filemode = false
> 	bare = false
> 	logallrefupdates = true
> 	ignorecase = true
> 
> [remote "origin"]
> 	url = codecommit://my-profile@repos
> 	fetch = +refs/heads/*:refs/remotes/origin/*
> 
> [branch "main"]
> 	remote = origin
> 	merge = refs/heads/main
> 
> [credential]
> 	UseHttpPath = true
> 	helper = !aws --profile my-profile codecommit credential-helper $@
> 

Hopefully this can be solved, as the goal of this specific credential manager is to _not_ have to save longterm credentials on the local machine.
